src/main/java/com/direwolf20/buildinggadgets/common/schema/template/provider/ITemplateProvider.java

```diff
@@ -0,0 +1,45 @@
+package com.direwolf20.buildinggadgets.common.schema.template.provider;
+
+import com.direwolf20.buildinggadgets.common.schema.template.Template;
+import jdk.internal.jline.internal.Nullable;
+import net.minecraftforge.fml.network.PacketDistributor.PacketTarget;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Consumer;
+
+public interface ITemplateProvider {
```

Comment:
What's the intention for this class?

Reply:
You said you came to like my world cap provides templates System... This is a simplified version of that and would be exposed just like that as a cap on world. (with the Client and the Server implementations which are still empty.

Reply:
Client for the render?

Reply:
Render and TemplateManager.

Reply:
Ah cool, I'll be intrested to see how you implement this.